apps/web/src/actions/user-profile-mod.ts

```diff
@@ -3,48 +3,186 @@
 import { authenticatedAction } from "@/lib/safe-action";
 import { z } from "zod";
 import { db } from "db";
-import { userCommonData } from "db/schema";
+import { userCommonData, userHackerData } from "db/schema";
 import { eq } from "db/drizzle";
 import { put } from "@vercel/blob";
 import { decodeBase64AsFile } from "@/lib/utils/shared/files";
 import { revalidatePath } from "next/cache";
-import { getUser } from "db/functions";
+import { getUser, getUserByTag } from "db/functions";
+import { RegistrationSettingsFormValidator } from "@/validators/shared/RegistrationSettingsForm";
 
-// TODO: Add skill updating
 export const modifyRegistrationData = authenticatedAction(
+	RegistrationSettingsFormValidator,
+	async (
+		{
+			age,
+			gender,
+			race,
+			ethnicity,
+			isEmailable,
+			university,
+			major,
+			levelOfStudy,
+			schoolID,
+			hackathonsAttended,
+			softwareBuildingExperience,
+			heardAboutEvent,
+			shirtSize,
+			dietaryRestrictions,
+			accommodationNote,
+			github,
+			linkedin,
+			personalWebsite,
+			phoneNumber,
+			countryOfResidence,
+		},
+		{ userId },
+	) => {
+		const user = await db.query.userCommonData.findFirst({
+			where: eq(userCommonData.clerkID, userId),
+		});
+		if (!user) throw new Error("User not found");
+		await db
```

Comment:
Optimization: I don't think these two requests depend on each other. Therefore, this is a good opportunity to parallelize these requests using `Promise.all()`. A good example of this can be found [here](https://nextjs.org/learn/dashboard-app/fetching-data#parallel-data-fetching)